cli/args.go

```diff
@@ -0,0 +1,136 @@
+package cli
+
+import (
+	"fmt"
+	"os"
+	"github.com/urfave/cli"
+	"github.com/gruntwork-io/terragrunt/options"
+	"github.com/gruntwork-io/terragrunt/config"
+	"github.com/gruntwork-io/terragrunt/errors"
+	"github.com/gruntwork-io/terragrunt/util"
+	"strings"
+)
+
+// Parse command line options that are passed in for Terragrunt
+func ParseTerragruntOptions(cliContext *cli.Context) (*options.TerragruntOptions, error) {
+	return parseTerragruntOptionsFromArgs(cliContext.Args())
+}
+
+// TODO: replace the urfave CLI library with something else.
+//
+// EXPLANATION: The normal way to parse flags with the urfave CLI library would be to define the flags in the
+// CreateTerragruntCLI method and to read the values of those flags using cliContext.String(...),
+// cliContext.Bool(...), etc. Unfortunately, this does not work here due to a limitation in the urfave
+// CLI library: if the user passes in any "command" whatsever, (e.g. the "apply" in "terragrunt apply"), then
+// any flags that come after it are not parsed (e.g. the "--foo" is not parsed in "terragrunt apply --foo").
+// Therefore, we have to parse options ourselves, which is infuriating. For more details on this limitation,
+// see: https://github.com/urfave/cli/issues/533. For now, our workaround is to dumbly loop over the arguments
+// and look for the ones we need, but in the future, we should change to a different CLI library to avoid this
+// limitation.
```

Comment:
+1 for https://github.com/mitchellh/cli. Although it's a little clunkier to use than urfave/cli, I believe all the HashiCorp tools are based on this library and it does a nice job of creating a pleasant end-user experience (including colors, nice formatting/tabs, etc.)